src/gc/gcee.cpp

```diff
@@ -405,7 +404,41 @@ void ProfScanRootsHelper(Object** ppObject, ScanContext *pSC, uint32_t dwFlags)
     }
 #endif //INTERIOR_POINTERS
     ScanRootsHelper(&pObj, pSC, dwFlags);
-#endif //  defined(FEATURE_EVENT_TRACE)
+#endif //  defined(GC_PROFILING) || defined(FEATURE_EVENT_TRACE)
+}
+
+// TODO - at some point we would like to completely decouple profiling
```

Comment:
The proper factoring to use for these is to move it these calls behind GCToEEInterface - so that the GC code does not contain calls to CLR/CoreCLR-specific COM interfaces, etc.

(What you are doing here is fine as a targeted fix for the bug.)